acir/src/circuit/opcodes.rs

```diff
@@ -30,125 +22,32 @@ pub enum BrilligOutputs {
     Array(Vec<Witness>),
 }
 
-impl BrilligOutputs {
-    fn to_u16(&self) -> u16 {
-        match self {
-            BrilligOutputs::Simple(_) => 0,
-            BrilligOutputs::Array(_) => 1,
-        }
-    }
-}
-
 #[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Debug)]
 pub struct Brillig {
     pub inputs: Vec<BrilligInputs>,
     pub outputs: Vec<BrilligOutputs>,
+    // results of oracles/functions external to brillig like a database read
+    pub foreign_call_results: Vec<ForeignCallResult>,
     pub bytecode: Vec<brillig_bytecode::Opcode>,
     /// Predicate of the Brillig execution - indicates if it should be skipped
     pub predicate: Option<Expression>,
 }
 
 impl Brillig {
     pub fn write<W: Write>(&self, mut writer: W) -> std::io::Result<()> {
-        let inputs_len = self.inputs.len() as u32;
-        write_u32(&mut writer, inputs_len)?;
```

Comment:
The bytecode format could just be specced as whatever the serde JSON of the rust is. It's practical and can always be customized later with custom directives I believe.